homeassistant/components/velux/config_flow.py

```diff
@@ -106,9 +124,55 @@ async def async_step_user(
                     data=user_input,
                 )
 
-        data_schema = self.add_suggested_values_to_schema(DATA_SCHEMA, user_input)
+        if not self.hosts:
+            aiozc = await zeroconf.async_get_async_instance(self.hass)
+            vd: VeluxDiscovery = VeluxDiscovery(zeroconf=aiozc)
```

Comment:
I'm not sure about this, do we really need both user triggered and automatic discovery?
Maybe we could add functionality to the Home Assistant `zeroconf` integration which allows fetching services already discovered for a services, similar to what we support for bluetooth: https://github.com/home-assistant/core/blob/4ff64534d37dfd472a23ad3ee58d074ee41bf93e/homeassistant/components/dormakaba_dkey/config_flow.py#L63

Reply:
I think adding that to zeroconf would be helpful long term. Right now users tend to ignore discoveries and than can’t find the integration and than open issues if they have no way to add it manually once it’s ignored. If they do discover that and unignore it, a restart is generally required to find it again.

When we solve these issues we probably don’t need manual adding anymore but we are still a ways off from a solution there. In the mean time it makes sense to implement manual adding via a user flow

Reply:
IMHO, we shouldn't add this here; and instead solve the issue at head instead.

Reply:
I agree. This PR could drive the needed change in the `zeroconf` integration. @pawlizio Is this something you could look into? You can contact me on Discord if it's not clear where to start.